op-batcher/batcher/channel_builder.go

```diff
@@ -107,9 +117,33 @@ func (c *channelBuilder) Blocks() []*types.Block {
 func (c *channelBuilder) Reset() error {
 	c.blocks = c.blocks[:0]
 	c.frames = c.frames[:0]
+	c.timeout = 0
+	c.fullErr = nil
 	return c.co.Reset()
 }
 
+// FramePublished calculates the submission timeout of this channel from the
+// given frame inclusion L1-block number. If an older frame tx has already been
+// seen, the timeout is not updated.
+func (c *channelBuilder) FramePublished(l1BlockNum uint64) {
+	timeout := l1BlockNum + c.cfg.ChannelTimeout - c.cfg.SubSafetyMargin
+	c.updateTimeout(timeout)
+}
+
+// TimedOut returns whether the passed block number is after the channel timeout
+// block. If no block timeout is set yet, it returns false.
+func (c *channelBuilder) TimedOut(blockNum uint64) bool {
+	return c.timeout != 0 && blockNum >= c.timeout
+}
+
+// TriggerTimeout checks if the channel is timed out at the given timestamp and
+// in this case sets the channel as full with reason ErrChannelTimedOut.
```

Comment:
```suggestion
// TriggerTimeout checks if the channel is timed out at the given block number and
// if it is timed out, marks the channel as full with reason ErrChannelTimedOut.
```

Reply:
IDK if I have a better name & I'm fine merging without changing it, but `TriggerTimeout` is slightly confusing to me because it doesn't always mark the channel as timed out.

Reply:
Agree, it's a confusing name. Renamed it to `CheckTimeout`.